src/Tools/dotnet-format/Program.cs

```diff
@@ -0,0 +1,103 @@
+using System.CommandLine.Invocation;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Tools.CodeFormatter.Logging;
+using Microsoft.CodeAnalysis.Tools.CodeFormatter.MSBuild;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.CodeAnalysis.Tools.CodeFormatter
+{
+    class Program
+    {
+        static async Task<int> Main(string[] args)
+        {
+            var parser = new CommandLineBuilder()
+                .UseParseDirective()
+                .UseHelp()
+                .UseSuggestDirective()
+                .UseParseErrorReporting()
+                .UseExceptionHandler()
+                .AddOption(new[] { "-w", "--workspace" }, "The project or solution to format", a => a.WithDefaultValue(() => null).ExactlyOne())
+                .AddOption(new[] { "-q", "--quiet" }, "Suppresses all output except warnings and errors", a => a.ParseArgumentsAs<bool>((sr) => ArgumentParseResult.Success(true), new ArgumentArityValidator(0, 0)))
+                .AddOption(new[] { "-v", "--verbose" }, "Show verbose output", a => a.ParseArgumentsAs<bool>((sr) => ArgumentParseResult.Success(true), new ArgumentArityValidator(0, 0)))
+                .AddVersionOption()
+                .OnExecute(typeof(Program).GetMethod(nameof(Run)))
+                .Build();
+
+            return await parser.InvokeAsync(args);
+        }
+
+        public static async Task<int> Run(string workspace, bool quiet, bool verbose, IConsole console = null)
+        {
+            var serviceCollection = new ServiceCollection();
+            var logLevel = GetLogLevel(quiet, verbose);
+            ConfigureServices(serviceCollection, console, logLevel);
+
+            var serviceProvider = serviceCollection.BuildServiceProvider();
+            var logger = serviceProvider.GetService<ILogger<Program>>();
+
+            try
+            {
+                if (quiet && verbose)
+                {
+                    logger.LogError(Resources.Error_QuietAndVerboseSpecified);
+                    return 1;
+                }
+
+                using (var cancellationTokenSource = new CancellationTokenSource())
+                {
+                    var workingDirectory = Directory.GetCurrentDirectory();
+                    var (isSolution, workspacePath) = MSBuildWorkspaceFinder.FindWorkspace(workingDirectory, workspace);
+
+                    MSBuildEnvironment.ApplyEnvironmentVariables();
+
+                    return await CodeFormatter.FormatWorkspaceAsync(logger, workspacePath, isSolution, cancellationTokenSource.Token);
+                }
+            }
+            catch (FileNotFoundException fex)
+            {
+                logger.LogError(fex.ToString());
+                return 1;
+            }
+            catch (Exception ex)
+            {
+                if (ex is TaskCanceledException || ex is OperationCanceledException)
```

Comment:
TaskCanceledException is already an OperationCanceledException.  So it's redundant to have both checks.